integration/src/adjudication/services/basket-cleanup-service/basket-cleanup.service.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { CUSTOM_OBJECT_CONTAINER_BASKET_STORE } from '../../../common/constants/constants';
+import { CustomObjectService } from '../../../common/providers/commercetools/custom-object/custom-object.service';
+import * as moment from 'moment';
+
+@Injectable()
+export class BasketCleanupService {
+  private containerKey = CUSTOM_OBJECT_CONTAINER_BASKET_STORE;
+  private logger = new Logger(BasketCleanupService.name);
+  private limit = this.configService.get<number>(
+    'storedBasketCleanup.objectQueryLimit',
+  );
+  private olderThanValue = this.configService.get<number>(
+    'storedBasketCleanup.olderThanValue',
+  );
+  private olderThanUnit = this.configService.get<string>(
+    'storedBasketCleanup.olderThanUnit',
+  );
+
+  constructor(
+    private customObjectService: CustomObjectService,
+    private configService: ConfigService,
+  ) {}
+
+  async clearOldBaskets(): Promise<any> {
+    let oldCustomObjects: { key: string; lastModifiedAt: string }[] = [];
+    let hasMore = false;
+    const olderThanDate = moment
+      .utc()
+      .subtract(this.olderThanValue as any, this.olderThanUnit as any);
+    this.logger.log(
+      `Getting saved baskets up to ${olderThanDate} (older than ${this.olderThanValue} ${this.olderThanUnit} ago)`,
+    );
+    const results = {
+      successful: [],
+      failed: [],
+    };
+    do {
+      const customObjects = await this.customObjectService.queryCustomObjects({
+        queryArgs: {
+          withTotal: false,
+          limit: this.limit,
+          container: this.containerKey,
+          offset: oldCustomObjects.length,
+          where: `lastModifiedAt < "${olderThanDate.toISOString()}"`,
+          sort: ['lastModifiedAt asc'],
+        },
+      });
+      hasMore = Boolean(customObjects.body.count === this.limit);
+      oldCustomObjects = oldCustomObjects.concat(
```

Comment:
I think we should delete directly all the custom objects here, I know is a very edge scenario but if the cronjob never ran for years there might be millions of objects to process and it might allocate too much memory to store all of them into oldCustomObjects